sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/MethodParameterUnused.CSharp8.cs

```diff
@@ -1,31 +1,74 @@
-﻿namespace SonarAnalyzer.UnitTest.TestCases
+﻿namespace Tests.TestCases
 {
-    public class WithLocalFunctions
+    public interface IWithDefaultImplementation
     {
-        public void Method()
+        decimal Count { get; set; }
+        decimal Price { get; set; }
+
+        void Reset(int a);     //Compliant
+
+        //Default interface methods
+        decimal Total()
+        {
+            return Count * Price;
+        }
+
+        decimal Total(decimal Discount)
+        {
+            return Count * Price * (1 - Discount);
+        }
+
+        decimal Total(string unused)    // Compliant, because it's interface member
+        {
+            return Count * Price;
+        }
+    }
+
+    public class StaticLocalFunctions
+    {
+        public int DoSomething(int a)   // Compliant
         {
-            void M1(int a,
-                    int b, // Noncompliant
-                    int c,
-                    int d) // Noncompliant
-                    {
-                        var result = a + c;
-                    }
```

Comment:
Could we keep 1 of these tests from master to show that we correctly raise the issue when several noncompliant parameters are found on the same method ?